addons/analytic/static/src/components/analytic_distribution/analytic_distribution.js

```diff
@@ -507,9 +507,14 @@ export class AnalyticDistribution extends Component {
     }
 
     onSaveNew() {
-        this.openTemplate({ resId: false, context: {
-            'default_analytic_distribution': this.listForJson,
-        }});
+        this.openTemplate({
+            resId: false, context: {
+                'default_analytic_distribution': this.listForJson,
+                'default_partner_id': this.props.record.data['partner_id'][0],
+                'default_product_id': this.props.record.data['product_id'][0],
+                'default_account_prefix': this.props.record.data['account_id'][1].substr(0, 3),
```

Comment:
Keep in mind the analytic distribution is used in many views, some of which will not have these fields available and will therefore cause a traceback. You could test this on the Sales Order. 
2 props `product_field` and `account_field` will give you an indication of whether those fields are available or you'd need to check the record/model as that would be necessary for partner.